sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobClient.java

```diff
@@ -0,0 +1,203 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.storage.blob;
+
+import com.azure.android.core.http.ServiceCallBack;
+import com.azure.android.core.http.ServiceClient;
+import com.azure.android.core.http.interceptor.AddDateInterceptor;
+import com.azure.android.core.internal.util.serializer.SerializerFormat;
+import com.azure.android.storage.blob.models.BlobItem;
+import com.azure.android.storage.blob.models.ContainersListBlobFlatSegmentResponse;
+import com.azure.android.storage.blob.models.ListBlobsIncludeItem;
+import com.azure.android.storage.blob.models.ListBlobsOptions;
+
+import java.util.List;
+import java.util.Objects;
+
+import okhttp3.Interceptor;
+
+/**
+ * Client for Storage Blob service.
+ */
+public class StorageBlobClient {
+    private final ServiceClient serviceClient;
+    private final StorageBlobServiceImpl storageBlobServiceClient;
+
+    private StorageBlobClient(ServiceClient serviceClient) {
+        this.serviceClient = serviceClient;
+        this.storageBlobServiceClient = new StorageBlobServiceImpl(this.serviceClient);
+    }
+
+    /**
+     * Creates a new {@link Builder} with initial configuration copied from this {@link StorageBlobClient}.
+     *
+     * @return A new {@link Builder}.
+     */
+    public StorageBlobClient.Builder newBuilder() {
+        return new Builder(this);
+    }
+
+    /**
+     * Gets the blob service base URL.
+     *
+     * @return The blob service base URL.
+     */
+    public String getBlobServiceUrl() {
+        return this.serviceClient.getBaseUrl();
+    }
+
+    /**
+     * Gets a list of blobs identified by a page id in a given container.
+     *
+     * @param pageId Identifies the portion of the list to be returned.
+     * @param containerName The container name.
+     * @param options The page options.
+     * @return A list of blobs.
+     */
+    public List<BlobItem> getBlobsInPage(String pageId,
+                                         String containerName,
+                                         ListBlobsOptions options) {
+        return this.storageBlobServiceClient.getBlobsInPage(pageId, containerName, options);
+    }
+
+    /**
+     * Gets a list of blobs identified by a page id in a given container.
+     *
+     * @param pageId Identifies the portion of the list to be returned.
+     * @param containerName The container name.
+     * @param options The page options.
+     * @param callback Callback that receives the retrieved blob list.
+     */
+    public void getBlobsInPage(String pageId,
+                               String containerName,
+                               ListBlobsOptions options,
+                               ServiceCallBack<List<BlobItem>> callback) {
```

Comment:
Info: We've both sync and async methods. I found call-back based async is natural to use with android framework APIs such as pagination.